src/app/containers/PageHandlers/withVariant/index.jsx

```diff
@@ -1,50 +1,41 @@
-import PropTypes from 'prop-types';
-import pathOr from 'ramda/src/pathOr';
 import React from 'react';
 import { useParams, useLocation, Redirect } from 'react-router-dom';
-import getVariantRedirectUrl from './getVariantRedirectUrl';
+import { variantSanitiser, getVariant } from '#lib/utilities/variantHandler';
+import {
+  getPreferredVariant,
+  setPreferredVariant,
+} from '#contexts/UserContext/cookies';
+
+const redirect = (location, service, defaultVariant) => {
+  const preferredVariant = getPreferredVariant(service);
+  return (
+    <Redirect
+      to={{
+        ...location,
+        pathname: `${location.pathname}/${preferredVariant || defaultVariant}`,
+      }}
+    />
+  );
+};
 
-const WithVariant = Component => {
+const withVariant = Component => {
   const VariantContainer = props => {
     const { service, variant } = useParams();
     const location = useLocation();
+    const defaultVariant = getVariant({ service, variant });
+    const sanitizedVariant = variantSanitiser(variant);
 
-    const { path, params } = pathOr({}, ['match'], props);
-
-    const redirectPath = getVariantRedirectUrl({
-      path,
-      params,
-      service,
-      variant,
-    });
-
-    if (!redirectPath) {
-      return <Component {...props} />;
+    // If no variant in path and service has a default variant which isn't 'default'.
+    if (!sanitizedVariant && defaultVariant && defaultVariant !== 'default') {
```

Comment:
this code smells a bit funny e.g. [commenting](https://github.com/ryanmcdermott/clean-code-javascript#only-comment-things-that-have-business-logic-complexity) (even the comment is hard to undertstand) and [hard to read conditional](https://github.com/ryanmcdermott/clean-code-javascript#encapsulate-conditionals). Also I don't think you need to check for the existence of `defaultVariant` before checking it's value.

Something like this would be better
```js
if (noVariantInPath && isNotDefaultVariant)  {
```